connector/solarwindsentityconnector/internal/event.go

```
@@ -15,47 +15,135 @@
+	relationshipLog, err := e.createRelationshipEvent(relationship, resourceAttrs)
+	if err != nil {
+		e.Logger.Debug("Failed to create relationship event", zap.Error(err))
 		return
 	}
 
-	dest, ok := entities[relationship.Destination]
+	relationshipLog.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+	relationshipLog.Attributes().PutStr(entityEventType, relationshipUpdateEventType)
+	eventLog := e.ResultEventLog.AppendEmpty()
+	relationshipLog.CopyTo(eventLog)
+
+}
+
+func (e *EventBuilder) createRelationshipEvent(relationship config.Relationship, resourceAttrs pcommon.Map) (plog.LogRecord, error) {
+	source, ok := e.Entities[relationship.Source]
 	if !ok {
-		zap.L().Debug("destination entity not found", zap.String("entity", relationship.Destination))
-		return
+		return plog.NewLogRecord(), fmt.Errorf("bad source entity")
 	}
 
-	event, err := CreateRelationshipEvent(resourceAttrs, relationship, src, dest)
-	if err != nil {
-		zap.L().Debug("failed to create relationship update event", zap.Error(err))
-		return
+	dest, ok := e.Entities[relationship.Destination]
+	if !ok {
+		return plog.NewLogRecord(), fmt.Errorf("bad destination entity")
 	}
 
-	event.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
-	event.Attributes().PutStr(entityEventType, relationshipUpdateEventType)
+	lr := plog.NewLogRecord()
+	attrs := lr.Attributes()
 
-	eventLog := lrs.AppendEmpty()
-	event.CopyTo(eventLog)
+	if source.Type == dest.Type {
+		// same type relationships
+		hasPrefixSrc, err := setIdAttributesSameType(attrs, source.IDs, resourceAttrs, relationshipSrcEntityIds, e.SourcePrefix)
+		if err != nil {
+			return plog.NewLogRecord(), fmt.Errorf("bad source entity id attributes")
+		}
+
+		hasPrefixDst, err := setIdAttributesSameType(attrs, dest.IDs, resourceAttrs, relationshipDestEntityIds, e.DestPrefix)
+		if err != nil {
+			return plog.NewLogRecord(), fmt.Errorf("bad destination entity id attributes")
+		}
+
+		if !hasPrefixSrc || !hasPrefixDst {
```

Maybe I'm missing something, but since we use or anyway and work with no particular info afterwards, would it make to fail fast? maybe propagate the `!hasPrefix` state as an error instead of separate bool value?